store/store.go

```diff
@@ -69,6 +70,19 @@ const (
 	UbuntuCoreWireProtocol = "1"
 )
 
+type rateLimitKey string
+
+func ContextWithRateLimit(ctx context.Context, rate float64) context.Context {
```

Comment:
sounds like we need a new DownloadOptions as well

Reply:
Yeah, I can add them and then we don't need the rate limit context anymore.

Reply:
Sure, adding this.